gradle-consistent-versions-idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/GradleCacheExplorer.java

```diff
@@ -82,10 +75,13 @@ public final Set<String> getCompletions(DependencyGroup input) {
         return Collections.emptySet();
     }
 
-    private Set<String> extractStrings(ProgressIndicator indicator) {
+    private Set<String> extractStrings(Set<String> repoUrls, ProgressIndicator indicator) {
```

Comment:
this is not a safe change, if you add a url to `repoUrls` you're not going to be able to see updated results for at most 10 mins as you've keyed on metadata.

it doesn't form part of your cache key

I think it's fine for now since it's all constant, but we're going to have to think about the caching strategy when you're reading urls from your xml file.

roughly speaking, perhaps you want to cache just extracting all the strings from file regardless of validity, and then post-process with the urls outside of this method. If that then becomes a hot path, we can think about caching again there, but it's a bit awkward. Thoughts @CRogers ?

Reply:
it could even be a 'cache invalidate on gradle reload' which is kinda aligned with what you want i.e. 
* you potentially get a new `repositories.xml` when you reload gradle
* you potentially get a new binary file when you reload gradle

I suppose the only thing it doesn't capture is running gradle outside of intellij and seeing results in , idk

Reply:
I've reverted these changes so that we can merge then we can figure this out in the PR that add the repo reading

Reply:
Invalidating when reloading gradle makes sense to me... probably also worth seeing just how long this takes to read.